### PR TITLE
Add AggregatorRouter Contract use case

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,41 +1,41 @@
...
-        "prettier": "^2.8.4",
-        "solhint": "^3.3.8",
-        "solhint-plugin-prettier": "^0.0.5",
-        "solidity-coverage": "^0.8.2",
-        "ts-node": "^10.9.1",
-        "typechain": "^8.1.1",
-        "typescript": "^4.9.4"
-    }
+  "name": "data-feed-consumer",
+  "version": "0.2.0",
```
Comment (martinkersner) [inline package.json]: I think we should also increase version here.

File: hardhat.config.ts [hunk 2]
```diff
@@ -7,15 +7,17 @@ import dotenv from 'dotenv'
 
 dotenv.config()
 
-let commonConfig = { gas: 5_000_000, gasPrice: 250_000_000_000 }
+let commonConfig = {}
```
Comment (martinkersner) [inline hardhat.config.ts:10]: Is there a reason to move definition inside the conditional branch?

File: contracts/DataFeedConsumer.sol [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
+import {IAggregatorRouter} from "@bisonai/orakl-contracts/src/v0.1/interfaces/IAggregatorRouter.sol";
 
 contract DataFeedConsumer {
     IAggregator internal dataFeed;
+    IAggregatorRouter internal router;
     int256 public answer;
     uint80 public roundId;
 
-    constructor(address aggregatorProxy) {
+    constructor(address aggregatorProxy, address aggregatorRouter) {
```
Comment (martinkersner) [inline contracts/DataFeedConsumer.sol]: I think passing `AggregatorProxy` loses its purpose after having access to the `AggregatorRouter`. How having two consumer contracts? One showing how to use `AggregatorProxy` and one for `AggregatorRouter`?

File: contracts/DataFeedConsumer.sol [hunk 4]
```diff
@@ -1,24 +1,27 @@
...
     }
 
     function getLatestData() public {
         (
             uint80 roundId_,
-            int256 answer_
-            , /* uint startedAt */
-            , /* uint updatedAt */
-            , /* uint80 answeredInRound */
+            int256 answer_ /* uint startedAt */ /* uint updatedAt */ /* uint80 answeredInRound */,
```
Comment (martinkersner) [inline contracts/DataFeedConsumer.sol]: Can we keep the previous format? 🙏

**Review by martinkersner — APPROVED** [review]
LGTM! I left couple comments :)

File: contracts/DataFeedConsumer.sol (hunk 4 above)
Comment (ghost) [reply contracts/DataFeedConsumer.sol]: it's automatically formatted from my vs code... I'll turn off format on save option and update it

File: hardhat.config.ts (hunk 2 above)
Comment (ghost) [reply hardhat.config.ts:10]: there were error from my environment so I had to change it..
image

File: contracts/DataFeedConsumer.sol (hunk 3 above)
Comment (ghost) [reply contracts/DataFeedConsumer.sol]: I felt bothered to make another contract and make separate deploy script, and thought that this is enough explanation for other developers to understand. But I'll write another contract if you insist

Comment (martinkersner) [reply contracts/DataFeedConsumer.sol]: Understood! We should assume that some people might want to take our code and make minimal modifications. In this case, they have to understand both approaches and always remove one of them. Please separate those two approaches into two consumer contracts.

File: hardhat.config.ts (hunk 2 above)
Comment (martinkersner) [reply hardhat.config.ts:10]: It complains about `accounts`, not about `gas` and `gasPrice`. Did you try to define accounts as below?

```typescript
let commonConfig = { gas: 5_000_000, gasPrice: 250_000_000_000, accounts: [] }
```

Then we could just update the `accounts`

```typescript
commonConfig.accounts = [process.env.PRIV_KEY]
```

Comment (ghost) [reply hardhat.config.ts:10]: don't know what's different from my environment but it still leads to error
Screenshot 2024-01-10 at 2 59 50 PM
image

Comment (ghost) [reply hardhat.config.ts:10]: image

Comment (martinkersner) [reply hardhat.config.ts:10]: You don't have to waste time on it. We can fix it later ;)

**Pull request merged**